server/src/main/webapp/WEB-INF/rails/webpack/views/pages/elastic_agents/wizard.tsx

```diff
@@ -0,0 +1,618 @@
+        <div>
+          <div class={styles.profileForm}>
+            <div><TextField label="Cluster Profile Name"
+                            readonly={vnode.attrs.etag !== undefined || vnode.attrs.readonly}
+                            property={vnode.attrs.clusterProfile().id}
+                            errorText={vnode.attrs.clusterProfile().errors().errorsForDisplay("id")}
+                            required={true}/></div>
+
+            <div><SelectField label="Plugin ID"
+                              property={this.pluginIdProxy.bind(this, vnode)}
+                              required={true}
+                              readonly={vnode.attrs.readonly}
+                              errorText={vnode.attrs.clusterProfile().errors().errorsForDisplay("pluginId")}>
+              <SelectFieldOptions selected={vnode.attrs.clusterProfile().pluginId()}
+                                  items={pluginList}/>
+            </SelectField>
+            </div>
+          </div>
+          <div>
+            <div
+              className={foundationClassNames(foundationStyles.foundationGridHax, foundationStyles.foundationFormHax)}>
+              <div class="row collapse" data-test-id="properties-form">
+                {this.clusterProfileForm(vnode)}
+              </div>
+            </div>
+          </div>
+        </div>
+        <ConceptDiagram image={clusterProfileImg} css={conceptDiagramStyles}>
+          <h2>What is a Cluster Profile?</h2>
+          <p>A Cluster Profile is the connection configuration of the environment where Elastic Agents run. Typically,
+            this includes the cluster connection URL, credentials, network, permission settings etc. Eg: Kubernetes
+            Cluster Configurations.</p>
+        </ConceptDiagram>
+      </div>
+    );
+  }
+
+  private pluginIdProxy(vnode: m.Vnode<Attrs>, newValue?: string) {
+    if (newValue) {
+      // if newValue is different from current value
+      if (this.pluginInfoForCurrentClusterProfileOrDefaultToFirstPlugin(vnode).id !== newValue) {
+        const newPluginInfo = this.findPluginWithId(vnode, newValue);
+
+        vnode.attrs.clusterProfile().pluginId(newPluginInfo!.id);
+        vnode.attrs.clusterProfile().properties(new Configurations([]));
+
+        vnode.attrs.elasticProfile().pluginId(newPluginInfo!.id);
+        vnode.attrs.elasticProfile().properties(new Configurations([]));
+      }
+    }
+    return vnode.attrs.clusterProfile().pluginId();
+  }
+
+  private pluginInfoForCurrentClusterProfileOrDefaultToFirstPlugin(vnode: m.Vnode<Attrs>) {
+    let result;
+    if (vnode.attrs.clusterProfile() && vnode.attrs.clusterProfile().pluginId()) {
+      const pluginId = vnode.attrs.clusterProfile().pluginId();
+      result         = this.findPluginWithId(vnode, pluginId);
+    }
+
+    if (!result) {
+      result = vnode.attrs.pluginInfos()[0];
+    }
+
+    return result;
+  }
+
+  private findPluginWithId(vnode: m.Vnode<Attrs>, pluginId?: string) {
```

Comment:
Suggestion:

We can use `findByPluginId` method on `PluginInfos` instead of this.